### ISSUE TITLE
Issue with video seekable only as a multiplier of 20 seconds

### ISSUE THREAD

**Issue by mzielezny** [legokichi/ts-ebml#10]
Hello,
I'm developing solution which is capturing video from Canvas with usage of Media Recorder API. 
Due to files produced by it are not seekable, I have tried to use your solution to resolve my issues. 
It is working and made seekable files but on Chrome in version 60 I have experienced that time of video is only set as multiplier of 20 seconds. If I made 13 seconds record in VLC video is unseekable (on bar 0 is shown), and if I made 25 seconds video, time is set only  to 20 seconds. In VLC when it play 20 seconds it is always playing rest 5 seconds but they are unseekable. 
It doesn't matter which codec I use (vp8/vp9) it always generate me the same issue. 
Do you have maybe any idea what could happen ?

**Comment by legokichi** [comment]
related #8

**Comment by legokichi** [comment]
chorme 59 EBML

* add MaxBlockAdditionID AlphaMode 
* https://bugs.chromium.org/p/chromium/issues/detail?id=690968
* https://codereview.chromium.org/2691373005
* https://codereview.chromium.org/2691373005/diff/300001/media/muxers/webm_muxer.cc
* https://matroska.org/technical/specs/index.html#BlockAdditional
* 
http://wiki.webmproject.org/alpha-channel

**Comment by legokichi** [comment]
* https://codereview.chromium.org/search?closed=1&owner=+emircan&reviewer=&cc=&repo_guid=&base=&project=&private=1&commit=1&created_before=&created_after=&modified_before=&modified_after=&order=&format=html&keys_only=False&with_messages=False&cursor=&limit=30
* https://codereview.chromium.org/search?closed=1&owner=vignesh&reviewer=&cc=&repo_guid=&base=&project=&private=1&commit=1&created_before=&created_after=&modified_before=&modified_after=&order=&format=html&keys_only=False&with_messages=False&cursor=&limit=30
* https://codereview.chromium.org/search?keys_only=False&cursor=CkcKFQoIbW9kaWZpZWQSCQj44Kri4__RAhIqahdzfmNocm9taXVtY29kZXJldmlldy1ocnIPCxIFSXNzdWUYq_-QgAoMGAAgAQ%3D%3D&format=html&cc=&with_messages=False&project=chromium&repo_guid=&base=&limit=30&owner=mcasas%40chromium.org&order=

**Comment by legokichi** [comment]
try https://github.com/legokichi/ts-ebml/releases/tag/v2.0.0

**Comment by mzielezny** [comment]
Sadly issue still occurs - when I record over 1 minute of video I could seek in first part but then after time of video it still playing "lost" part, and I could seek if record is shorter than even 30 seconds.

**Comment by Venryx** [comment]
Yeah, I get the same issue with the video sometimes having a "lost" part that plays after the listed duration is reached. (I don't have the issue of being unable to seek for <30s videos, though)

Anyway, my program was in Electron, so I opted to just use ffmpeg for the add-seekability task.

`ts-eblm` works fairly well overall, but if it's easy to do (ie. your program is in electron so has access to utils like ffmpeg), one might as well use those utils since they don't have the offset issue. (plus ffmpeg doesn't have the "thumbnail not getting generated in Windows explorer" issue, and presumably it also runs a bit faster, though I haven't confirmed this)
